lib/gather/commands/show.rb

```diff
@@ -42,7 +42,28 @@ def run
           data = data.deep_merge(Collector.logical_data)
           File.open(Config.data_path, 'w') { |file| file.write(data.to_yaml) }
         end
-        File.open(Config.data_path) { |file| puts file.read }
+
+        data = YAML.load_file(Config.data_path)
+        keys = (@args[0] || '.').delete_prefix('.').split('.')
+
+        value = data
+        keys.each do |key|
+          if key.match(/\[[^\]]*\]/)
```

Comment:
Please could you add a comment here explaining what this block is for? I can see that it's removing excess square brackets when they exist, but when would they exits/why don't they _always_ exist?

Reply:
In `jq` syntax square brackets are used to specify keys which aren't symbols, for example `.items.[1]` would get the first value from the array under `items`, whereas `.items.1` would look for a key called `1` under the field `items`. The same applies for strings; `.[my_key]` will look for the key `"my_key"`, whereas `.my_key` will look for the key `:my_key`.

So if square brackets are found, then their contents should be taken and converted into a string or integer.